src/main/java/org/gnieh/logback/config/ConfigConfigurator.java

```diff
@@ -76,51 +77,32 @@ public void configure(LoggerContext loggerContext) {
 			} else {
 				addWarn("Invalid ROOT logger configuration. Ignoring it.");
 			}
-
 		}
 
 		Config loggerConfigs = config.getConfig("loggers");
 		for (Entry<String, ConfigValue> entry : loggerConfigs.root().entrySet()) {
 			if (entry.getValue() instanceof ConfigObject) {
-				configureLogger(loggerContext, appenders, entry.getKey(), loggerConfigs.getConfig(entry.getKey()),
-						false);
+				configureLogger(loggerContext, appenders, entry.getKey(), loggerConfigs.getConfig("\"" + entry.getKey() + "\""), false);
 			} else {
 				addWarn(String.format("Invalid logger configuration %s. Ignoring it.", entry.getKey()));
 			}
 		}
-
 	}
 
-	private Appender<ILoggingEvent> configureAppender(LoggerContext loggerContext, String name, Config config)
-			throws InstantiationException, IllegalAccessException, ClassNotFoundException {
+	private Appender<ILoggingEvent> configureAppender(LoggerContext loggerContext, String name, Config config) throws ReflectiveOperationException {
+		List<Object> children = new ArrayList<>();
 
-		Class<?> clazz = Class.forName(config.getString("class"));
 		@SuppressWarnings("unchecked")
-		Appender<ILoggingEvent> appender = (Appender<ILoggingEvent>) clazz.newInstance();
+		Class<Appender<ILoggingEvent>> clazz = (Class<Appender<ILoggingEvent>>) Class.forName(config.getString("class"));
 
+		Appender<ILoggingEvent> appender = this.configureObject(loggerContext, clazz, config, children);
 		appender.setName(name);
 
-		// for each key appearing in the appender configuration, check whether
-		// it is a bean property and call the correct method on the appender
-		ConfigPropertySetter propertySetter = new ConfigPropertySetter(beanCache, appender);
-		for (String key : config.withoutPath("class").root().keySet()) {
-			if ("encoder".equals(key)) {
-				Encoder<?> encoder = configureEncoder(loggerContext, config.getConfig("encoder"));
-				propertySetter.setProperty(key, encoder);
-			} else if ("layout".equals(key)) {
-				Layout<?> layout = configureLayout(loggerContext, config.getConfig("layout"));
-				propertySetter.setProperty(key, layout);
-			} else if ("filter".equals(key)) {
-				Filter<?> filter = configureFilter(loggerContext, config.getConfig("filter"));
-				propertySetter.setProperty(key, filter);
-			} else if ("filters".equals(key)) {
-				for (Config c : config.getConfigList("filters")) {
-					Filter<?> filter = configureFilter(loggerContext, c);
-					propertySetter.setProperty(key, filter);
-				}
-			} else {
-				propertySetter.setProperty(key, config);
-			}
+		for (Object child : children) {
+			if (child instanceof RollingPolicy)
```

Comment:
Please put braces even if there is a single statement in the then branch

Reply:
done